spec/models/badge_achievement_spec.rb

```diff
@@ -18,8 +21,12 @@
     expect(achievement.rewarding_context_message).to include("</a>")
   end
 
-  it "awards credits after create" do
-    expect(achievement.user.credits.size).to eq(5)
+  it "doesn't award credits id credits_awarded is zero" do
```

Comment:
_Quick, non-blocking suggestion!:_
```suggestion
  it "doesn't award credits if credits_awarded is zero" do
```